### PR TITLE
CKAN 2.10

### PR DESCRIPTION
**Pull request by nickumia-reisys**
Related to
- https://github.com/GSA/data.gov/issues/4209

Changes:
- `IRoutes` is deprecated and it wasn't doing anything for us since `CKAN 2.9.5`
- (BREAKING) Update references to routes from pylons `organization_read` to new way of `organization.read` ... et cetera.
- Default testing suite to `CKAN 2.10`
- (breaking) Update tests to use `SysadminWithToken` instead of `Sysadmin` since `apikey` has been deprecated in favor of `token`
- (BREAKING) SQLAlchemy deprecated `._row` (https://github.com/GSA/ckanext-usmetadata/pull/70/commits/4a76eb02d557287aaf8c160ae8c2a41dbd43f7c8)
- Update validators from `ckan.plugins.toolkit.get_validator` to `ckan.lib.navl.validators`
- Create custom validator for `string`
- Remove custom template helper `usmetadata_shorten` (functionality taken care of by `truncate`)
- Update template forms to reference `{{ h.csrf_input() }}`
- CKAN deprecated `truncate` so use builtin jinja function instead...
  - https://jinja.palletsprojects.com/en/3.1.x/templates/#jinja-filters.truncate
  - https://github.com/ckan/ckan/pull/6299
- Specify `CKAN__SITE_LOGO` for local testing
- Update ADR with decision to remove `datapreview` block
- Update `package/resource_read.html` template to load metadata dict properly (`labels` key was removed in `2.10`)

### CODE REVIEW

**Comment by nickumia-reisys** [comment]
Tests are passing.. TODO:
- [ ] Update README
- [ ] Sync `CKAN 2.10` templates
- [ ] Any other random changes... TBD

**Comment by btylerburton** [comment]
Added a commmit to remove the call to core for `data_preview` block. This fixes the inventory resource display issue.

**Comment by nickumia-reisys** [comment]
@btylerburton The README definitely still needs to be updated.  I don't know of any other critical template updates left.  So generally, this is ready for release.  I was going to delay extension releases until we were done with integration tests on catalog and inventory.  If we are happy with those and we are ready to move to staging and prod, then yes, we can merge this and make a release.

File: README.md [hunk 1]
```diff
@@ -24,13 +20,12 @@ This extension is compatible with these versions of CKAN.
 
 CKAN version | Compatibility
 ------------ | -------------
-<=2.8        | no
-2.9          | yes
+>=2.10       | yes
```
Comment (nickumia-reisys) [inline README.md:23]:
```suggestion
<=2.9        | no
2.10         | yes
```

**Comment by FuhuXia** [comment]
What is with the `2.10.01`? openknowledge/ckan-dev only got `2.10.1`.

**Review by FuhuXia — APPROVED** [review]

**Pull request merged**